examples/lib/stories/components/components_notifier_example.dart

```diff
@@ -0,0 +1,103 @@
+import 'package:flame/components.dart';
+import 'package:flame/game.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+class ComponentsNotifierExampleWidget extends StatefulWidget {
+  const ComponentsNotifierExampleWidget({super.key});
+
+  static String description = '''
+''';
+
+  @override
+  State<ComponentsNotifierExampleWidget> createState() =>
+      _ComponentsNotifierExampleWidgetState();
+}
+
+class _ComponentsNotifierExampleWidgetState
+    extends State<ComponentsNotifierExampleWidget> {
+  @override
+  void initState() {
+    super.initState();
+
+    game = ComponentNotifierExample();
+  }
+
+  late final ComponentNotifierExample game;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: MultiProvider(
+        providers: [
+          Provider<ComponentNotifierExample>.value(value: game),
+          ChangeNotifierProvider<ComponentsNotifier<Enemy>>(
+            create: (_) => game.componentsNotifier<Enemy>(),
+          ),
+        ],
+        child: Stack(
+          children: [
+            Positioned.fill(
+              child: GameWidget(game: game),
+            ),
+            const Positioned(
+              left: 16,
+              top: 16,
+              child: GameHud(),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class GameHud extends StatelessWidget {
+  const GameHud({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final enemies = context.watch<ComponentsNotifier<Enemy>>().components;
+
+    return Card(
+      child: Padding(
+        padding: const EdgeInsets.all(16),
+        child: enemies.isEmpty
+            ? ElevatedButton(
+                child: const Text('Play again'),
+                onPressed: () {
+                  context.read<ComponentNotifierExample>().replay();
+                },
+              )
+            : Text('Remaining enemies: ${enemies.length}'),
+      ),
+    );
+  }
+}
+
+class Enemy extends CircleComponent with Tappable, Notifiable {
+  Enemy({super.position})
+      : super(
+          radius: 20,
+          paint: Paint()..color = const Color(0xFFFF0000),
+        );
+
+  @override
+  bool onTapUp(_) {
+    removeFromParent();
+    return true;
+  }
+}
```

Comment:
I'd add an example of listening to changes of custom properties. Because I expect that's what most folks will be doing most of the time.

Reply:
Good call, I've added a snippet covering that on the docs.